main/wallet-backend.js

```
@@ -69,119 +83,160 @@ class WalletBackend {
-      const amountMinusGas = amount.minus(gasLimit).minus(gasOffset)
-      const message = new Message({
-        to,
-        from,
-        nonce: await this.provider.getNonce(from),
-        value: amountMinusGas.toAttoFil(),
-        method: 0,
-        params: ''
-      })
-      const messageWithGas = await this.provider.gasEstimateMessageGas(
-        message.toLotusType()
-      )
-      const lotusMessage = messageWithGas.toLotusType()
-      const msgValid = await this.provider.simulateMessage(lotusMessage)
-      assert(msgValid, 'Message is invalid')
-      const signedMessage = await this.provider.wallet.sign(from, lotusMessage)
-      const { '/': cid } = await this.provider.sendMessage(signedMessage)
-
+      const tx = await fn()
+      console.log({ tx })
+      assert(tx.hash, 'Transaction hash not found')
+      const cid = await this.convertEthTxHashToCid(tx.hash)
+      console.log({ cid })
       transaction.hash = cid
       this.onTransactionUpdate()
-
       return cid
     } catch (err) {
       console.error(err)
       transaction.status = 'failed'
       this.onTransactionUpdate()
-
       throw err
     }
   }
 
+  /**
+   * @param {string} hash
+   * @returns {Promise<string>}
+   */
+  async convertEthTxHashToCid (hash) {
+    console.log('convertEthTxHashToCid', { hash })
+    for (let i = 0; i < 10; i++) {
+      const res = await fetch('https://graph.glif.link/query', {
+        headers: {
+          accept: '*/*',
+          'accept-language': 'en-US,en;q=0.9,de;q=0.8',
+          'content-type': 'application/json'
+        },
+        body: JSON.stringify({
+          operationName: 'Message',
+          variables: {
+            cid: hash
+          },
+          query: 'query Message($cid: String!) {message(cid: $cid) {cid}}'
+        }),
+        method: 'POST'
+      })
+      const body = await res.json()
+      if (body.data.message) {
+        return body.data.message.cid
+      }
+      console.log(body.errors)
```

Do we have Sentry integration available here? Should we report the error to Sentry in such case?

---

Errors here are expected, when the transaction can't yet be found. These aren't critical, only debug errors

---

I've changed to only log if we see an error here that we're not expecting